### PR TITLE
Add support for upscaling 

### PR DESCRIPTION
**Pull request by agapioscon**
Users can use from 2 models: 
(default) RealESRGAN or stable-diffusion-x4-upscaler
Optional parameters:
-output file
-scale
-prompt

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,6 +18,7 @@ lib/
 lib64/
 parts/
 sdist/
+weights/
```
Comment (trizko) [inline .gitignore]: Can you put this at the bottom of the file? I made a section at the bottom for non-standard .gitignore entries.

File: aictl/cli/main.py [hunk 2]
```diff
@@ -189,6 +189,52 @@ def t2a(args):
...
+            pipe.enable_model_cpu_offload()
+            pipe.enable_xformers_memory_efficient_attention()
+        pipe = pipe.to(device)
+        upscaled_image = pipe(prompt=args.prompt, image=image).images[0]
+        upscaled_image.save(args.output_path)
+
+    if args.model == "esrgan":
+        model = RealESRGAN(device, scale=args.scale)
+        model.load_weights(
+            "weights/RealESRGAN_x" + str(args.scale) + ".pth", download=True
```
Comment (trizko) [inline aictl/cli/main.py]: We use fstrings in the rest of the codebase and we should keep it consistent here.

File: aictl/cli/main.py [hunk 3]
```diff
@@ -403,6 +449,46 @@ def main():
     )
     t2a_parser.set_defaults(func=t2a)
 
+    ####Upscale
+    upscale_parser = subparsers.add_parser("upscale", help="Upscale an image")
+    upscale_parser.add_argument("-p", "--prompt", default="", help="the prompt to use")
```
Comment (trizko) [inline aictl/cli/main.py]: In the help section of this argument, let's make it explicit that this argument is only available for the stable diffusion model.

File: aictl/cli/main.py [hunk 4]
```diff
@@ -403,6 +449,46 @@ def main():
...
+        "-o",
+        "--output-path",
+        default=f"output_upscale{utc_time}.png",
+        help="the path for audio when generation is complete",
+    )
+    upscale_parser.add_argument(
+        "-m",
+        "--model",
+        default="esrgan",
+        help="the upscale model (x4) to use (options: esrgan,sdx4:use this on your own risk.OOM danger!!)",
```
Comment (trizko) [inline aictl/cli/main.py]: You can remove the `:use this on your own risk.OOM danger!!` part. This whole repo has danger of OOM haha

File: aictl/cli/main.py [hunk 5]
```diff
@@ -189,6 +189,52 @@ def t2a(args):
...
+        if is_mac:
+            pipe.enable_attention_slicing()
+        else:
+            pipe.enable_model_cpu_offload()
+            pipe.enable_xformers_memory_efficient_attention()
+        pipe = pipe.to(device)
+        upscaled_image = pipe(prompt=args.prompt, image=image).images[0]
+        upscaled_image.save(args.output_path)
+
+    if args.model == "esrgan":
```
Comment (trizko) [inline aictl/cli/main.py]: I know it won't make a difference but logically speaking this should be an `elif`

**Review by trizko — COMMENTED** [review]
Awesome feature to add. I left a few comments.

File: .gitignore (hunk 1 above)
Comment (agapioscon) [reply .gitignore]: Done

File: aictl/cli/main.py (hunk 2 above)
Comment (agapioscon) [reply aictl/cli/main.py]: Done

File: aictl/cli/main.py (hunk 3 above)
Comment (agapioscon) [reply aictl/cli/main.py]: done

File: aictl/cli/main.py (hunk 4 above)
Comment (agapioscon) [reply aictl/cli/main.py]: done

File: aictl/cli/main.py (hunk 5 above)
Comment (agapioscon) [reply aictl/cli/main.py]: done

File: setup.py [hunk 6]
```diff
@@ -30,6 +30,9 @@
             'pytest',
             'ruff',
         ],
+        'upscaler': [
+            'RealESRGAN @ git+https://github.com/ai-forever/Real-ESRGAN.git',
```
Comment (trizko) [inline setup.py]: I missed this part in my last review but I think this can actually go into the `install_requires` section.

**Comment by trizko** [comment]
Thank you!

**Pull request merged**